### PR TITLE
Improvements to release procedure

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- link to doc of GPG key preparation
- more detail about preparing release notes

### CODE REVIEW

File: content/apt/developers/release/maven-project-release-procedure.apt [hunk 1]
```diff
@@ -60,6 +62,38 @@ Performing a Maven Project Release
...
+
+  Generally it is a good that project use tha latest available versions of used dependencies.
+
+* Prepare release notes
+
+ * check all commits since latest release, all mentioned issue should be connected with release notes
+
+ * all issues should be fixed and <<CLOSED>>
+
+ * all issues should be <<assigned>> to responsible person, which fixed or merged fix
```
Comment (hgschmie) [inline content/apt/developers/release/maven-project-release-procedure.apt:75]: change the order? assign first, then close?

File: content/apt/developers/release/maven-project-release-procedure.apt [hunk 2]
```diff
@@ -60,6 +62,38 @@ Performing a Maven Project Release
...
+
+ []
+
+** Check release note at GitHub
+
+  If a component has a history of release notes at <<GitHub>>, we should to continue maintain it in order to avoid user confusing.
+
+  We can maintain release notes at <<GitHub>>, by one of:
+
+  * use release-drafter to help prepare release notes - in such case all PRs should a proper label
```
Comment (hgschmie) [inline content/apt/developers/release/maven-project-release-procedure.apt:87]: example?

**Review by hgschmie — APPROVED** [review]
Thank you for improving documentation!

File: content/apt/developers/release/maven-project-release-procedure.apt (hunk 2 above)
Comment (olamy) [reply content/apt/developers/release/maven-project-release-procedure.apt:87]: just look at this https://github.com/apache/maven-compiler-plugin/releases/tag/maven-compiler-plugin-3.11.0
each PR have an appropriate label/category so there are ordered

File: content/apt/developers/release/maven-project-release-procedure.apt (hunk 1 above)
Comment (slawekjaranowski) [reply content/apt/developers/release/maven-project-release-procedure.apt:73]: done

**Comment by cstamas** [comment]
Just FYI I did for myself a "note" (w/ commands to run) reminder, maybe is good to have them on this page as well? 
https://github.com/apache/maven-indexer/blob/master/RELEASE.md

**Review by cstamas — APPROVED** [review]

**Review by olamy — APPROVED** [review]

**Review by slachiewicz — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
> Just FYI I did for myself a "note" (w/ commands to run) reminder, maybe is good to have them on this page as well? https://github.com/apache/maven-indexer/blob/master/RELEASE.md
> 
> Edit: typically it is **good thing** (and is my practice) to run `mvn -P reporting site site:stage` BEFORE actual release, as I had cases where _site generation failed after release (ie. due javadoc issue)_.

for site we have a step `Make sure that site compilation works` with command

other commands can be added in next PR 😄

**Pull request merged**